File: Circulat Queue/circularqueue.py
class Queue:

    def __init(self, queue, size:int = 0):

        self.queue = queue
        self.front = -1
        self.rear = -1
        if size == 0:
            self.size = len(queue)
        elif size < 0:
            print("------------Size must be positive------------\n")
        else:
            self.size = size

    def is_empty(self):

        if self.front == -1 and self.rear == -1 :
            return True
        
        return False
    
    def is_full(self):

        if self.front == 0 and self.rear == self.size-1 :
            return True
        
        if self.front == self.rear+1 and self.rear != -1:
            return True
        
        return False
    
    def enqueue(self, val):

        if self.is_empty() :
            front = rear = -1
            front+=1
        
        if self.is_full():
            print("Queue is full")

        else:

            if self.rear == self.size -1:
                self.rear = -1

            self.rear += 1
            self.queue[self.rear] = val

    def dequeue(self):

        if self.is_empty():
            print("Queue is empty")

        else:
            if ( self.front == self.rear ):
                self.front = -1
                self.rear = -1
            else:
                self.front += 1
        


        

File: Circulat Queue/circularqueue.cpp
#include <iostream>
#define size 10
using namespace std;

class queue{

    private:
        int queue_[size];
        int front, rear;

    public:

        queue(){
            front = -1; rear = -1;
        }

        bool is_empty(){
            if(front==-1&&rear==-1){
                return true;
            }
            return false;
        }

        bool is_full(){
            if(front == 0 && rear == size - 1){
                cout<<rear;
                return true;
            }
            if(front == rear+1 && rear != -1){
                return true;
            }
            return false;
        }

        void enqueue(int val){
            if (is_empty()){
                front = rear = -1;
                front++;
            }
            if(is_full()){
                cout<<"Queue is full";
            }
            else{

                if (rear == size){
                    rear = -1;
                }

                rear++;
                queue_[rear] = val;
            }
        }

        void enqueue(int arr[], int n = size){
            if (is_empty()){
                front = rear = -1;
                front++;
            }
            if(is_full()){
                cout<<"Queue is full";
            }
            else{
                for ( int x=0; x < n; x++ ){
                    rear++;
                    queue_[rear] = arr[x];
                }
            }
        }

        void dequeue(){
            if(is_empty()){
                cout<<"Queue is empty";
            }
        
            else{
                front++;
                if ( front==rear+1 ){
                front = rear = -1;
                }
            }
             
        }

        void display(){
            for(int i = front; i<=rear; i++){
                cout<<queue_[i];
            }
        }

};



int main(){

    queue q1;
    int arr[10] = {1,2,3,4,5,6,7,8,9, 10};
    q1.enqueue(arr, 8);
    q1.display();


    return 0;
}